datalad_gooey/tests/test_fsbrowser_item.py

```diff
@@ -1,7 +1,160 @@
-
 from pathlib import Path
+
 from ..fsbrowser_item import FSBrowserItem
 
+from datalad.distribution.dataset import Dataset
+from datalad.tests.utils_pytest import (
+    assert_equal,
+    assert_in,
+    assert_not_in,
+    assert_raises,
+    with_tempfile,
+
+)
+from PySide6.QtCore import (
+    Qt,
+)
+
+from ..status_light import GooeyStatusLight
+
+
+@with_tempfile
+def test_GooeyStatusLight(path=None):
+    ds = Dataset(path).create()
+    GooeyStatusLight.__call__(dataset=ds, path=ds.pathobj)
+
 
 def test_FSBrowserItem():
-    FSBrowserItem(Path.cwd())
+    fsb_item = FSBrowserItem(Path.cwd())
+    assert_equal(fsb_item.data(0, FSBrowserItem.PathObjRole), Path.cwd())
+    assert_equal(fsb_item.__str__(), f'FSBrowserItem<{Path.cwd()}>')
+
+
+def test_pathobj_none():
+    with assert_raises(RuntimeError):
+        FSBrowserItem(None).pathobj()
+
+
+item_types = ['directory', 'symlink', 'file', 'dataset']
+lsdir_results = [
+    {
+        'path': f'some_{t}.txt',
+        'type': t,
+        'action': 'gooey-lsdir',
+        'status': 'ok'
+    }
+    for t in item_types
+]
+lsdir_results+=\
+    [
+        {
+            'path': 'test_error_no_message',
+            'type': 'file',
+            'action': 'gooey-lsdir',
+            'status': 'error',
+            'message': ''
+        },
+        {
+            'path': 'test_error_with_message',
+            'type': 'file',
+            'action': 'gooey-lsdir',
+            'status': 'error',
+            'message': 'Permissions denied'
+        },
+    ]
+def test_update_from_lsdir_result():
+    for res in lsdir_results:
+        fsb_item = FSBrowserItem(res['path'])
+        fsb_item.update_from_lsdir_result(res)
+        # check if item type is same as result type
+        assert_equal(fsb_item.data(1, Qt.EditRole), res['type'])
```

Comment:
```suggestion
        assert_equal(fsb_item.datalad_type, res['type'])
```

Same rational as above.